docs/operating-eck/configure-operator-metrics.asciidoc

```diff
@@ -0,0 +1,398 @@
+
+=== Using the ECK Operator Helm Chart
+
+If you installed ECK through the Helm chart commands listed in <<{p}-install-helm>>, you can now set `config.metrics.secureMode.enabled` to `true` and both RBAC and TLS/HTTPs will be enabled for the metrics endpoint.
+
+==== Using your own TLS certificate for the metrics endpoint when using the Helm Chart
+
+By default a self-signed certificate will be generated for use by the metrics endpoint. If you want to use your own TLS certificate for the metrics endpoint you can provide the `config.metrics.secureMode.tls.certificateSecret` to the Helm chart. The `certificateSecret` should be the name of an existing Kubernetes `Secret` that contains both the TLS certificate and the TLS private key. The following keys are supported within the secret:
+
+* `tls.crt` - The PEM-encoded TLS certificate
+* `tls.key` - The PEM-encoded TLS private key
+
+The easiest way to create this secret is to use the `kubectl create secret tls` command. For example:
+
+[source,sh]
+----
+kubectl create secret tls eck-metrics-tls-certificate -n elastic-system --cert=/path/to/tls.crt --key=/path/to/tls.key
+----
+
+Providing this secret is sufficient to use your own certificate if it is from a trusted Certificate Authority. If the certificate is not signed by a trusted CA you have 2 options:
+
+* Disable TLS verification.
+    * Set `config.metrics.secureMode.tls.insecureSkipVerify` to `true` to disable TLS validation.
+* Provide the Certificate Authority to Prometheus.
+    * Set `config.metrics.secureMode.tls.insecureSkipVerify` to `false` to enable TLS validation.
+    * Set `config.metrics.secureMode.tls.caSecret` to the name of an existing Kubernetes secret within the Prometheus namespace that contains the CA in PEM format.
+    * Set the `spec.secrets` field of the `Prometheus` custom resource such that the CA secret is mounted into the Prometheus pod at `config.metrics.secureMode.tls.caMountDirectory` (assuming you are using the Prometheus operator). See the link:{eck_github}/tree/{eck_release_branch}/deploy/eck-operator/values.yaml[ECK Helm Chart values file] for more information.
+
+See the <<{p}-prometheus-requirements,prometheus requirements section>> for more information on creating the CA secret.
+
+=== Using the ECK Operator manifests
+
+If you installed ECK through using the manifests using the commands listed in <<{p}-deploy-eck>> some additional changes will be required to enable secure metrics.
+
+* Enable the metrics port in the `ConfigMap`, and set the metrics host to `127.0.0.1` to force communication through `kube-rbac-proxy`.
+
+[source,sh]
+----
+kubectl get cm elastic-operator -n elastic-system -o yaml | sed "s|metrics-port: 0|metrics-port: 8081|" | sed "s|metrics-host: 0.0.0.0|metrics-host: 127.0.0.1|" | kubectl apply -f -
```

Comment:
can we maybe try to construct a `kubectl patch` command to update this ?

Reply:
Unfortunately that was the first thing I tried. Since this is a string in the configmap, it just doesn't work that way. 😞 Suggestions appreciated to handle this cleaner...

Reply:
ooh right!! I think then this is the best option AFAIK.

Reply:
Updated to include a full configmap apply instead.